app/routes/api/maintainer.ts

```diff
@@ -0,0 +1,56 @@
+import { ActionFunction, json } from "@remix-run/node";
+import { getUserByUid } from "~/database.server";
+import { getProjectsMaintainedByUser } from "~/projects.server";
+
+/**
+ * This API endpoint verifies that a user is listed as the maintainer of at
+ * least 1 project in OSH. It expects a `userId` in the payload and a valid
+ * bearer token.
+ */
+export const action: ActionFunction = async ({ request }) => {
+  // Check that we have a valid userId in the payload
+  const data = await request.json();
+  const { userId } = data;
+  if (typeof userId !== "string") {
+    return new Response(null, {
+      status: 500,
+      statusText: "Malformed request",
+    });
+  }
+
+  // Check that we have a valid API key
```

Comment:
This endpoint will be called by the Discord bot, and because I can't predict what URL it will call from, I can't really rely on CORS. So I set up a basic API key system to restrict who can call this endpoint.